torch/_decomp/decompositions.py

```diff
@@ -579,6 +579,30 @@ def native_dropout_backward(grad_output: Tensor, mask: Tensor, scale: float):
     return grad_output * (mask.type_as(grad_output) * scale)
 
 
+@register_decomposition(aten.reciprocal)
+def reciprocal(self: Tensor) -> Tensor:
+    return 1 / self
+
+
+# Used for implementing ops that implicitly promote ints to floats. Could be in Primtorch refs?
+def _promote_to_float(self: Tensor) -> Tensor:
+    if self.dtype == torch.float64:
+        return self
+    return self.to(torch.float)
+
+
+@register_decomposition(aten.logit)
+def logit(self: Tensor, eps: Optional[float] = None) -> Tensor:
+    self = _promote_to_float(self)
+    if eps is None:
+        eps = -1.0
+    lo = eps
+    hi = 1 - eps
+    self = torch.clamp(self, lo, hi)
+    print(self.dtype)
```

Comment:
pront

Reply:
So what is it for half inputs? Looks like it should be float, and you'd return float that you didn't expect?